torch/distributed/_tensor/device_mesh.py

```diff
@@ -143,6 +143,13 @@ def __init__(
                 f"Mesh should not be bigger than default world size, but found {self.mesh.numel()} ranks!"
             )
 
+        # TODO: we will support mesh on a subset of WORLD in future
+        if self.mesh.numel() < world_size:
```

Comment:
this check should not happen all the time, it should only happen when there's no default pg exist and we want to help user create a world_pg, this check should be only inside get_or_create_group I think

Reply:
Make sense because IIRC we can define for example mesh A on rank 0, 1 and mesh B on rank 2, 3. The example we discussed last time is actually about mesh is defined on rank 0, 1 and no mesh is defined on rank 2, 3. Right?

Reply:
yeah, it's possibly to create sub meshes, that's what 2-D did currently, so we should still allow such behavior.